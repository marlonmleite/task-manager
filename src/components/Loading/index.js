import React from 'react'
import PropTypes from 'prop-types'
import LoadingStyled from './styled'

const Loading = ({ color }) => <LoadingStyled color={color}>Loading...</LoadingStyled>

Loading.defaultProps = {
  color: 'black',
}

Loading.propTypes = {
  color: PropTypes.string,
}

export default Loading
