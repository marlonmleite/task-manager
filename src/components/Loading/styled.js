import styled from 'styled-components'

const LoadingStyled = styled.span`
  color: ${({ color }) => color};
`

export default LoadingStyled
