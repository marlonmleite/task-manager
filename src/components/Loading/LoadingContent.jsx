import React from 'react'
import { Loading, LoadingContent as LoadingContentStyled } from './styled'

const LoadingContent = () => (
  <LoadingContentStyled>
    <Loading />
  </LoadingContentStyled>
)

export default LoadingContent
