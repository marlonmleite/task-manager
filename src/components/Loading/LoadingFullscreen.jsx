import React from 'react'
import { Loading, LoadingFullscreen as LoadingFullscreenStyled } from './styled'

const LoadingFullscreen = () => (
  <LoadingFullscreenStyled>
    <Loading />
  </LoadingFullscreenStyled>
)

export default LoadingFullscreen
