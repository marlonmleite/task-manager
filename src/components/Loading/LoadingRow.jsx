import React from 'react'
import { Loading, LoadingFullscreenStyled } from './styled'

const LoadingRow = () => (
  <LoadingFullscreenStyled background="rgba(255, 255, 255, 0.85)">
    <Loading size="20px" />
  </LoadingFullscreenStyled>
)

export default LoadingRow
