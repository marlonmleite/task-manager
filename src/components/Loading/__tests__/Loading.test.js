import React from 'react'
import { shallow } from 'enzyme'
import Loading from '../'
import LoadingStyled from '../styled'

describe('Loading', () => {
  it('should render', () => {
    const component = shallow(<Loading />)

    expect(component).toMatchSnapshot()
  })

  it('should color black', () => {
    const component = shallow(<LoadingStyled color="black" />)

    expect(component).toHaveStyleRule('color', 'black')
  })
})
