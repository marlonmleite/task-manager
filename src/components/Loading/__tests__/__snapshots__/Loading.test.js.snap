// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render 1`] = `
<styled.span
  color="black"
>
  Loading...
</styled.span>
`;
